src/custom-elements/src/components/date-picker/date-picker.tsx

```diff
@@ -338,27 +300,65 @@ export class DatePicker {
     }, CHI_TIME_AUTO_SCROLL_DELAY);
   }
 
-  @Listen('chiTimeChange')
-  handleTimeChange(ev) {
-    const chiDate = this.el.querySelector('.chi-popover__content chi-date');
-    let activeDate = chiDate.getAttribute('value');
+  handleDateChange(ev: CustomEvent) {
+    ev.stopPropagation();
 
-    if (!activeDate) {
-      const currentTime = new Date();
+    const date = ev.detail;
 
-      activeDate = `${currentTime.getMonth() + 1}/${currentTime.getDate()}/${currentTime.getFullYear()}`;
+    if (this.mode === 'date') {
+      this.value = date;
+      this._input.value = date;
     }
 
-    chiDate.setAttribute('value', activeDate);
-    if (this.timeFormat === '24hr') {
-      this.value = `${activeDate}, ${this.formatTimePeriod(ev.detail.hour)}:${this.formatTimePeriod(ev.detail.minute)}`;
-    } else {
-      const hour = ev.detail.hour > 12 ? ev.detail.hour - 12 : ev.detail.hour;
+    if (this.mode === 'datetime') {
+      if (!this.time) {
+        return;
+      }
```

Comment:
To avoid nested if's:

```suggestion
    if (this.mode === 'datetime' && this.time) {
```

Reply:
Done.